src/test/regress/expected/alter_index.out

```diff
@@ -0,0 +1,125 @@
+CREATE SCHEMA alterindex;
+SET search_path TO "alterindex";
+SET citus.next_shard_id TO 980000;
+SET client_min_messages TO WARNING;
+SET citus.shard_count TO 32;
+SET citus.shard_replication_factor TO 1;
+-- test alter index set statistics
+CREATE TABLE t1 (a int, b int);
+SELECT create_distributed_table('t1','a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE INDEX test_idx on t1 ((a+b));
+ALTER INDEX test_idx ALTER COLUMN 1 SET STATISTICS 4646;
+-- test alter index set statistics before distribution
+CREATE TABLE t2 (a int, b int);
+CREATE INDEX test_idx2 on t1 ((a+b), (a-b), (a*b));
+SELECT create_distributed_table('t2','a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+ALTER INDEX test_idx2 ALTER COLUMN 2 SET STATISTICS 3737;
+-- verify statistics is set
+SELECT c.relname, a.attstattarget
+FROM pg_attribute a
+-- WHERE attrelid IN (
+--     SELECT oid
+--     FROM pg_class
+--     WHERE relname LIKE 'test_idx%'
+-- )
+JOIN pg_class c ON a.attrelid = c.oid AND c.relname LIKE 'test_idx%'
+ORDER BY c.relname, a.attnum;
+  relname  | attstattarget
+---------------------------------------------------------------------
+ test_idx  |          4646
+ test_idx2 |            -1
+ test_idx2 |          3737
+ test_idx2 |            -1
+(4 rows)
+
+\c - - - :worker_1_port
+SELECT c.relname, a.attstattarget
+FROM pg_attribute a
+-- WHERE attrelid IN (
+--     SELECT oid
+--     FROM pg_class
+--     WHERE relname LIKE 'test_idx%'
+-- )
+JOIN pg_class c ON a.attrelid = c.oid AND c.relname LIKE 'test_idx%'
+ORDER BY c.relname, a.attnum;
+     relname      | attstattarget
+---------------------------------------------------------------------
+ test_idx2_980000 |            -1
+ test_idx2_980000 |          3737
```

Comment:
to reduce the verbosity here, could do something like:

```sql
--expect [x?]
SELECT COUNT(*)
FROM pg_attribute a
JOIN pg_class c ON a.attrelid = c.oid AND c.relname LIKE 'test_idx%'
ORDER BY c.relname, a.attnum
WHERE attstattarget=4646;

--expect [x?]
SELECT COUNT(*)
FROM pg_attribute a
JOIN pg_class c ON a.attrelid = c.oid AND c.relname LIKE 'test_idx2%'
ORDER BY c.relname, a.attnum
WHERE attstattarget=-1;

[and same with 3737]
```

Reply:
I think reducing shard_count could also work

Reply:
I think we should keep these rows because we can see which column target is altered for which shard, since they are ordered by relname and attnum. In tests, let's say we want to alter the stats target for column 2, we should see that the altered one is col2, not 1 or 3.

Reply:
Talked in person. Reduced shard count